### PR TITLE
Create book snippet

### PR DESCRIPTION
**Pull request by SarahhhC**
Create book snippet
* integrate all snippets when same book's snippet added.
* if book was not registered in firebase, search information => register needed. contained in TODO list.

### CODE REVIEW

File: src/server/graphQLschema/mutation/book.mutation.js [hunk 1]
```diff
@@ -15,6 +18,80 @@ import BookType from '../type/book.type';
...
+              author: val.author || null,
+              category: val.category || null,
+              thumbnail: val.thumbnail || null,
+              thumbnailColor: val.thumbnailColor || null,
+              publisher: val.publisher || null,
+              publishedDate: val.publishedDate || null,
+              crawledAt: val.crawledAt || null,
+              savedCount: val.savedCount || null,
+              isSaved: val.isSaved || null,
+              mySnippets: val.mySnippets || null,
```
Comment (festinalente91) [inline src/server/graphQLschema/mutation/book.mutation.js]: I think you don't have to save mySnippets and otherSnippets.
They are Graphql's schema, not firebase's.

File: src/server/graphQLschema/type/book.type.js [hunk 2]
```diff
@@ -64,7 +65,8 @@ const BookType = new GraphQLObjectType({
             .catch(reject);
         });
       }
-    }
+    },
+    snippets: { type: GraphQLJSON },
```
Comment (festinalente91) [inline src/server/graphQLschema/type/book.type.js:68]: is this required?

File: src/server/graphQLschema/mutation/book.mutation.js (hunk 1 above)
Comment (SarahhhC) [reply src/server/graphQLschema/mutation/book.mutation.js]: Ok, nice!

File: src/server/graphQLschema/type/book.type.js (hunk 2 above)
Comment (SarahhhC) [reply src/server/graphQLschema/type/book.type.js:68]: This one needed in firebase, so I contained this.
Can I ignore that?

Comment (festinalente91) [reply src/server/graphQLschema/type/book.type.js:68]: oh. I see.
By the way, shouldn't it be like new GraphQLList(GraphQLJSON) ?

Comment (SarahhhC) [reply src/server/graphQLschema/type/book.type.js:68]: `GraphQLList(GraphQLJSON)` makes some errors. I thought `GraphQLJSON` and `GraphQLList(GraphQLJSON)` are same but maybe they had some differences.

**Review by festinalente91 — APPROVED** [review]

**Pull request merged**